config/boards/olimex-teres-a64.conf

```diff
@@ -0,0 +1,36 @@
+# OLIMEX Teres-A64
+# Maintainer: Jacob "Kreyren" Hrbek <kreyren@fsfe.org>
+
+# Optimized for Mainboard Rev.C
+
+BOARD_NAME="OLIMEX Teres A64" # Allwinner A64 quad core 2GB SoC Wi-Fi/BT
+
+# Specified in https://linux-sunxi.org/Allwinner_SoC_Family
+BOARDFAMILY="sun50iw1"
+
+# Has custom default configuration of the u-boot that should be used -- https://linux-sunxi.org/Olimex_Teres-A64#Mainline_U-Boot
+BOOTCONFIG="teres_i_defconfig"
+
+# Observed issues on insufficient size, thus enforced -- https://github.com/OLIMEX/DIY-LAPTOP/issues/52
+BOOTSIZE="256"
+
+# Seems to be the most reliable and bootloader should stay as separate partition
+BOOTFS_TYPE="fat"
+
+# Legacy has been removed as it's too old and disfunctional
+KERNEL_TARGET="current,edge"
+
+# gpt was not tested
+IMAGE_PARTITION_TABLE="msdos"
+
+# Enable serial console by default
+DEFAULT_CONSOLE="serial"
+SERIALCON="ttyS0:115200"
+
+# Firmware realtek is needed for WiFi/BLE module
+# wpasupplicant is dependency of networkmanager to be able to scan for WiFi Access Points
+# Pulseaudio is needed for the audio to work especially in pipewire configuration
+PACKAGE_LIST_BOARD="firmware-realtek wpasupplicant pulseaudio"
```

Comment:
Why "firmware-realtek wpasupplicant " ? Firmware (if needed for wifi) that is needed for "bloated" armbian-firmware has to be added to armbian firmware repository which contains very small selection of firmware files that are present on SBCs (yes, some are certainly useless), but also for many popular USB based which we cover.

"pulseaudio" Why audio works without this elsewhere? And if this general package is solving things for this particular board, it might fix other similar boards too. This is our general philosophy. We always try to expand fix (if its clear) to as many similar devices as possible. I don't know if this is the case here, but its worth looking a bit.

Reply:
firmware-realtek provides the blobs without which the network-manager is unable to see WiFi access points and connect to them

wpasupplicant is dependency of NetworkManager service which fails if it's not provided with 'wlan0 couldn't initiate supplicant interface: failed to D-Bus activate wpa_supplicant service'
	* This is likely a bug in the build framework as wpasupplicant should be included with NetworkManager
	
> "pulseaudio" Why audio works without this elsewhere? 

I don't know and asked debian, but they don't know either.

The reproductors are wired directly to the A64 BGA
![teres-audio](https://user-images.githubusercontent.com/11302521/216702643-05650472-ee30-40e1-a44e-c019ec2bedce.png)

So in theory ALSA should be enough, but when tested in practice ALSA doesn't work and won't show up the relevant card in pipewire and on pulseaudio it needs `pactl set-card-profile alsa_card.platform-sound pro-audio` of using `pavucontrol` to change the card profile for pipewire to be able to use it instead of it being mistakenly taken as `Stereo Input`, documented on https://wiki.debian.org/InstallingDebianOn/Olimex/Teres-I#Audio

>  And if this general package is solving things for this particular board, it might fix other similar boards too. This is our general philosophy. We always try to expand fix (if its clear) to as many similar devices as possible. I don't know if this is the case here, but its worth looking a bit.

I didn't observe this issue on other boards and i have OLIMEX OlinuXino-A64 that i will be likely making port for as well so might investigate there more.

Reply:
CC @TsvetanUsunov in case you have any relevant info on the audio thing~

Reply:
The missing wpa_supplicant package may be a problem of debian sid/bookworm, at least I experienced it a couple of times on a fresh community build for OrangePi4 LTS. If so, please add wpa_supplicant to the packages required per OS version and not just for the board in another pull request, thanks.

For the realtek firmware, what @igorpecovnik is suggesting is that you should integrate the other repository just adding the firmware you need for the board, and not include another, potentially conflicting, package.

Reply:
> in case you have any relevant info on the audio thing

I think he has more important (like we all) things to do:
![EPrhOTnXkAEkoVL](https://user-images.githubusercontent.com/6281704/216761317-bbdd6fd4-07ef-4ee3-bdb5-796d6db913f9.jpeg)

We have been playing with this device 4 years ago:
https://www.youtube.com/watch?v=2I6cD5xF0FQ
and at one point everything worked, but if device is not maintained, it fells apart pretty quickly. This is Linux, our time is extremely limited which is why we are very pissed off when someone does not respect that ...

I have assembled, dissembled and assembled again but in last iteration keyboard connection broke and its in this state for year(s). Sadly I have 100+ devices, but also obligation to keep this project running and Tsvetan has his own company to run.

One of first builds: https://twitter.com/armbian/status/1009360529100591105 Some time after, it was also ported to Debian, so that you won't have stupid ideas of what is coming from Debian - user-land packages.

Reply:
> This is Linux, our time is extremely limited which is why we are very pissed off when someone does not respect that ... -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1096515547)

![annoying-me-again](https://user-images.githubusercontent.com/11302521/216775193-1100addc-7beb-4ea6-8be3-f8ba610d932f.png)

> I have assembled, dissembled and assembled again but in last iteration keyboard connection broke and its in this state for year(s). Sadly I have 100+ devices, but also obligation to keep this project running and Tsvetan has his own company to run. -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1096515547)

like if you don't want the system you can send it to https://wiki.hackerspaces.org/Base48 and i gift it to some motivated developer next to a guy who knows how to fix connectors~ (if you do then please sign the back of the mainboard with sharpie imma making a collection :3)

Reply:
> For the realtek firmware, what @igorpecovnik is suggesting is that you should integrate the other repository just adding the firmware you need for the board, and not include another, potentially conflicting, package. -- @paolosabatino (https://github.com/armbian/build/pull/4807#discussion_r1096511763)

Why though? It's stuff that can be outsourced on debian instead of adding maintainance overhead in armbian.. firmware packages rarely have any conflicts with other packages.

Reply:
> The missing wpa_supplicant package may be a problem of debian sid/bookworm, at least I experienced it a couple of times on a fresh community build for OrangePi4 LTS. If so, please add wpa_supplicant to the packages required per OS version and not just for the board in another pull request, thanks. -- @paolosabatino (https://github.com/armbian/build/pull/4807#discussion_r1096511763)

Addressed in [`d595ebc` (#4807)](https://github.com/armbian/build/pull/4807/commits/d595ebcd48e97ae4984673f6e150e65a3a8bd6b9) and running new build to confirm that solution implemented in future merge request is sufficient

Reply:
Merge request for wpasupplicant submitted @ https://github.com/armbian/build/pull/4814

Reply:
> Why though? It's stuff that can be outsourced on debian instead of adding maintainance overhead in armbian.. firmware packages rarely have any conflicts with other packages.

I can guess there may be several flexibility reasons:
- make the images smaller, not including all the firmwares from debian/ubuntu but just a selected number on mainly wifi/network firmwares;
- include firmwares that are not included in base debian for some reason (license is not very clear, general "slowness" in including binaries, ...);
- put files in some directories wanted by some drivers: drivers from legacy kernels often want non-common paths or filenames.

Overall, once established, I also guess the maintenance for the firmware repository is quite low.

Reply:
Would prefer to take a look at it as an additional merge request as i want to do baby steps in terms of improvement, but will look in like ~10h (when i wake up)

Reply:
> when i wake up

You have 3 unresolved comments which you need to resolve before moving on.

> in terms of improvement

In past week we had to close down around 80% of your contribution. Calm down!

Reply:
> You have 3 unresolved comments which you need to resolve before moving on. -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1096623287)

Comments:
* https://github.com/armbian/build/pull/4807#discussion_r1096235270 
* https://github.com/armbian/build/pull/4807#discussion_r1096234302 
* https://github.com/armbian/build/pull/4807#discussion_r1096231571

Are waiting on your reply for longer than 48h atm

Reply:
> In past week we had to close down around 80% of your contribution.  -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1096623287)

Merge Requests:
* https://github.com/armbian/build/pull/4742
* https://github.com/armbian/build/pull/4743
* https://github.com/armbian/build/pull/4744
* https://github.com/armbian/build/pull/4745

Were submitted due to wrong documentation and was told that approach like:
* https://github.com/armbian/build/pull/4746
"takes too much time to review" so once it was highlighted it was addressed in:
* https://github.com/armbian/build/pull/4748

https://github.com/armbian/build/pull/4750 and https://github.com/armbian/build/pull/4755 were closed despire having an important fixes for nix that are still complicating my life to this day

https://github.com/armbian/build/pull/4749 was moved in https://github.com/armbian/build/pull/4756 again due to the lack of proper documentation and management about submitting changes in `armbian-next` branch which was addressed (branch renamed and duplicate removed) when i pointed it out and issue was not re-openned so that i can address the highlighted issue thus had to make this one.

https://github.com/armbian/build/pull/4790 was closed despite such change being a good practice for a production environment to make sure that the CI/CD doesn't break on seemingly hard-coded name change

https://github.com/armbian/build/pull/4810 waits for reply

https://github.com/armbian/build/pull/4814 you didn't like my solution and instead of waiting for me to adjust the merge request you closed it resulting in me having to make a new one in https://github.com/armbian/build/pull/4817 -> Bad management through forcing multiple merge requests about the same issue.

That's 11 out of 23 none of which are my fault for reasons highlighted (possibly excl. 4810 due to waiting for reply).

If you dislike that there are closed merge requests then adjusts the management namely through adapting the practice of turning merge requests into drafts instead of closing them so that the contributor has time to apply the requested changes prior to re-submitting a review or don't complain about it as it seems to be designed this way to rush the review and ideally revisit the highlighted nix patches.

> Calm down and adjust to MO & the speed of others as others adjusting to yours is unreal to expect. -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1096623287)

As highlighted all your previous feedback in relation to my contributions were addressed to my best ability if you want additional changes then this merge request is not the place for it.. either make a new issue/forum post or talk with me on IRC.

Reply:
Firmware addressed in [`50f3e19` (#4807)](https://github.com/armbian/build/pull/4807/commits/50f3e195be7c53334de5819d6d964fa31e9a1aa7)

Reply:
Seems that highlighted issues were resolved